controllers/topics/create.js

```diff
@@ -0,0 +1,37 @@
+const {Topics} = require('../../databases/models');
+const {insertTopics} = require('../../utils/post');
+
+module.exports = async (req, res) => {
+  try {
+    let {name} = req.body;
+    name = name.toLowerCase();
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
+
+    if (topics) {
+      return res.status(400).json({
+        code: 400,
+        status: 'failed',
+        message: 'Topic already exists'
+      });
+    }
+
+    insertTopics([name]);
+
+    return res.status(200).json({
+      success: true,
+      message: 'Success'
+    });
+  } catch (error) {
+    return res.json({
+      code: error.statusCode,
+      status: 'fail',
+      message: error.message,
+      data: 'null'
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling is present which is good. However, it's better to send a specific HTTP status code instead of always sending 200 OK status, even in case of an error. You can use `res.status(error.statusCode || 500)` to send the error code if it's available, otherwise send a 500 Internal Server Error status.

```diff
-    return res.json({
+    return res.status(error.statusCode || 500).json({
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->